### DISCUSSION TITLE
Using CDN links in "demo" `web/viewer.html`

### DISCUSSION THREAD

**Discussion by IonMich** [mozilla/pdf.js#17642] [Q&A]
Is it possible to to get the `web/viewer.html` file to work properly with the suggested CDN distribution of `PDF.js`? Currently, there are four references to other files in the prebuilt version `web/viewer.html`:
```
<link rel="resource" type="application/l10n" href="locale/locale.json">
<script src="../build/pdf.mjs" type="module"></script>
<link rel="stylesheet" href="viewer.css">
<script src="viewer.mjs" type="module"></script>
```

I have already verified that if I change the reference of "../build/pdf.mjs" to:
```
<script src="https://cdn.jsdelivr.net/npm/pdfjs-dist@4.0.379/build/pdf.mjs" type="module"></script>
```
then the pdf still renders properly. 

But when I similarly try to change the link to the CSS file "viewer.css" to:
```
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/pdfjs-dist@4.0.379/web/pdf_viewer.css">
```
it no longer renders properly. Same for the `viewer.mjs`. I think this makes sense, because it is missing assets, like the icons. Since all these assets are still available through the CDN (see e.g. at [jsDelivr](https://www.jsdelivr.com/package/npm/pdfjs-dist?tab=files&path=web%2Fimages), I would expect to still be able to somehow import everything through CDN.

To summarize:
- is it possible to get the `web/viewer.html` working through CDN by merely changing the four file references to URL links in these four lines?
- if yes, is there a minimal working example of that somewhere in the documentation? Specifically for the references to "viewer.mjs" and "viewer.css". I can't find any. I think it would be useful to add something like that in the `Including via a CDN` section of the Wiki if the answer is non-trivial.

I also understand that according to the documentation one should re-skin the viewer before adding it to their website. So I understand that other parts of the `viewer.html` will not stay the same and that additional styling should be done.

**Comment by R-W-C** [comment]
Short answer: The library is on CDN's and the demo is not.

Long answer: There are two versions of pdf.js. Well actually there is only one, but you have got the library and you have got the demo.

First there is the library, which you can find here: https://mozilla.github.io/pdf.js 
Also there is the demo, which you can find here: https://github.com/mozilla/pdf.js#online-demo

There are many ways you can download the library, see the github page: 

- https://github.com/mozilla/pdf.js

If you download the zip file, you have got a file named pdf.js-master.zip 

The code of the demo (incl. library)  you can download here:  

- https://sourceforge.net/projects/pdf-js.mirror/files/latest/download
- https://mozilla.github.io/pdf.js/getting_started/#download 
- https://github.com/mozilla/pdf.js/releases

If you download the demo, you will have a file which looks like this: pdfjs-4.10.38-dist.zip

The library, pdf.mjs and some other files are hosted on different CDN's.  
One of these files is pdf_viewer.mjs. This is not the same file as viewer.mjs which is part of the demo and cannot be found on the CDN's!

Discaimer: I am not involved in the project, so I could be wrong.
